### PR TITLE
Add a linalg.pinv wrapper to common tensor

### PR DESCRIPTION
**Pull request by ljaljushkin**
### Changes
The following functions were added:

- linalg.py: pinv

### Reason for changes
Part of int4 with LoRA adaptation implementation.

### Related tickets
CVS-135863

### Tests
tests/shared/test_templates/template_test_nncf_tensor.py

### CODE REVIEW

File: tests/shared/test_templates/template_test_nncf_tensor.py [hunk 1]
```diff
@@ -1349,6 +1349,16 @@ def test_fn_linalg_inv(self, a, ref):
         assert fns.allclose(res.data, ref_tensor)
         assert res.device == tensor_a.device
 
+    def test_fn_linalg_pinv(self):
+        a = [[1.0], [2.0]]
+        A = Tensor(self.to_tensor(a))
+        B = fns.linalg.pinv(A)
+        print(B)
```
Comment (AlexanderDokuchaev) [inline tests/shared/test_templates/template_test_nncf_tensor.py:1356]:
```suggestion
```

File: nncf/tensor/functions/torch_linalg.py [hunk 2]
```diff
@@ -40,6 +40,19 @@ def _(a: torch.Tensor) -> torch.Tensor:
     return torch.linalg.inv(a)
 
 
+@linalg.pinv.register(torch.Tensor)
+def _(a: torch.Tensor) -> torch.Tensor:
+    """
+    Consider using torch.linalg.lstsq() if possible for multiplying a matrix on the left by the pseudo-inverse, as:
```
Comment (AlexanderDokuchaev) [inline nncf/tensor/functions/torch_linalg.py:46]: Why note from torch documentation added as docstring?
If you think that this note can be useful, add it as comment

**Review by AlexanderDokuchaev — CHANGES_REQUESTED** [review]

File: tests/shared/test_templates/template_test_nncf_tensor.py (hunk 1 above)
Comment (ljaljushkin) [reply tests/shared/test_templates/template_test_nncf_tensor.py]: Done

File: nncf/tensor/functions/torch_linalg.py (hunk 2 above)
Comment (ljaljushkin) [reply nncf/tensor/functions/torch_linalg.py]: Done

**Review by AlexanderDokuchaev — APPROVED** [review]

**Pull request merged**